brushfire-core/src/main/scala/com/stripe/brushfire/Lazy.scala

```diff
@@ -0,0 +1,36 @@
+package com.stripe.brushfire
+
+import com.twitter.algebird.{Monoid, Semigroup}
+
+/**
+  * Wrapper class to allow expensive [[Semigroup]] operations on a [[Tree]] to be deferred until first use
+  */
+class Lazy[A](fn: () => A) extends Serializable {
```

Comment:
Seems like this might be better to include in algebird directly? Doesn't seem to be a good fit in here, but it does provide compatibility with the way `SplitNode` used to behave for it's `lazy val annotation: A` field.